tests/test_helpers.nim

```diff
@@ -30,3 +32,20 @@ template procSuite*(name, body: untyped) =
       body
 
   suitePayload()
+
+type RngWrap = object
```

Comment:
What's the context for this? Could we add a brief comment here no why we need all this?

Reply:
So I am not quite sure how this works, maybe @dryajov can help. He used this in libp2p code which is where I got it from.

Reply:
So let's add a link to that piece of the code if we don't understand it. This means we can track any changes/rationale.